src/frontend/src/components/useSearchApartment.tsx

```diff
@@ -18,56 +23,113 @@ function useSearchBook(query: string, pageNum: number) {
 
    useEffect(() => {
       setApartments(array);
-   }, [query]);
+   }, [press, selected]);
 
+   const isInitialMount = useRef<boolean>(true);
    useEffect(() => {
-      setLoading(true);
-      setError(false);
-      const CancelToken = axios.CancelToken;
-      const source = CancelToken.source();
-      axios({
-         method: 'GET',
-         url: `http://localhost:3333/mockdata?q=${query}&_page=${pageNum}&_limit=2`,
-         params: { q: query, page: pageNum },
-         cancelToken: source.token,
-      })
-         .then((res) => {
-            const newApartments: {
-               name: string;
-               address: string;
-               image: string;
-               review: string;
-               rating: number;
-               price_min: string;
-               price_max: string;
-            }[] = [];
-            for (let i = 0; i < res.data.length; i++) {
-               newApartments.push({
-                  name: res.data[i].name,
-                  address: res.data[i].address,
-                  image: res.data[i].image,
-                  review: res.data[i].review,
-                  rating: res.data[i].rating,
-                  price_min: res.data[i].price_min,
-                  price_max: res.data[i].price_max,
+      if (isInitialMount.current == true || query === '') {
```

Comment:
This one deals with populating the mainpage right?

Reply:
I put in some clarifying comments.